packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -76,21 +106,40 @@ export abstract class AbstractPaymentService
     data: Data
   ): Promise<PaymentSessionData>
 
+  /**
+   * @param context The type of this argument is meant to be temporary and once the previous method signature
+   * will be removed, the type will only be PaymentContext instead of Cart & PaymentContext
+   */
+  public abstract createPayment(
+    context: Cart & PaymentContext
+  ): Promise<PaymentSessionResponse>
+
+  /**
+   * @deprecated use createPayment(context: Cart & PaymentContext): Promise<PaymentSessionResponse> instead
+   * @param cart
+   */
   public abstract createPayment(cart: Cart): Promise<PaymentSessionData>
-  public abstract createPaymentNew(
-    paymentInput: PaymentProviderDataInput
-  ): Promise<PaymentSessionData>
 
   public abstract retrievePayment(paymentData: PaymentData): Promise<Data>
 
+  /**
+   * @param paymentSessionData
+   * @param context The type of this argument is meant to be temporary and once the previous method signature
+   * will be removed, the type will only be PaymentContext instead of Cart & PaymentContext
+   */
   public abstract updatePayment(
     paymentSessionData: PaymentSessionData,
-    cart: Cart
-  ): Promise<PaymentSessionData>
-
-  public abstract updatePaymentNew(
+    context: Cart & PaymentContext
+  ): Promise<PaymentSessionResponse>
+
+  /**
+   * @deprecated use updatePayment(paymentSessionData: PaymentSessionData, context: Cart & PaymentContext): Promise<PaymentSessionResponse> instead
+   * @param paymentSessionData
+   * @param cart
+   */
+  public abstract updatePayment(
```

Comment:
line 130 has the same method defined, wouldn't this method override that?

Reply:
Since they have the same name, this is an overload, you can still implement one or the other but at least one is displayed as deprecated

<img width="652" alt="Screenshot 2022-12-16 at 15 43 15" src="https://user-images.githubusercontent.com/25098370/208122937-1299770c-5853-49cc-8cec-ec9f27019887.png">

Reply:
It can be seen roughtly like that

<img width="724" alt="Screenshot 2022-12-16 at 15 44 35" src="https://user-images.githubusercontent.com/25098370/208123184-2c646fb0-6e38-4f55-b8fd-19cf04aed03e.png">